lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioInputHelper.java

```diff
@@ -113,12 +110,9 @@ private int doInternalCacheRead(long position, byte[] bytes, int offset, int len
         int remainingLength = length;
         while (remainingLength > 0) {
             int bytesReadFromCache = readPageFromCache(position, bytes, offset, remainingLength);
-            if (bytesReadFromCache == 0) {
+            if (bytesReadFromCache <= 0) {
```

Comment:
Can we include a code comment explaining why ?

Reply:
We have added it.